ui/dashboards/src/components/GridLayout/GridLayout.tsx

```diff
@@ -75,11 +83,15 @@ export function GridLayout(props: GridLayoutProps) {
           containerPadding={[0, 10]}
           layouts={{ [GRID_LAYOUT_SMALL_BREAKPOINT]: groupDefinition.itemLayouts }}
           onLayoutChange={handleLayoutChange}
+          onWidthChange={handleWidthChange}
         >
-          {groupDefinition.itemLayouts.map(({ i }) => (
+          {groupDefinition.itemLayouts.map(({ i, w }) => (
             <div key={i}>
               <ErrorBoundary FallbackComponent={ErrorAlert}>
-                <GridItemContent panelGroupItemId={{ panelGroupId, panelGroupItemLayoutId: i }} />
+                <GridItemContent
+                  panelGroupItemId={{ panelGroupId, panelGroupItemLayoutId: i }}
+                  width={w * gridColWidth}
```

Comment:
calculating the panel width here so we get a more appropriate estimate for `useSuggestedStepMs`

Reply:
I did a little testing and I don't think the math here is quite right. The `width` you pass is what's console logged  in the screenshot, and it doesn't match the element width. I tried this with resizing a bunch to make sure I didn't hit a weird edge case.

<img width="1290" alt="image" src="https://user-images.githubusercontent.com/396962/231909035-55e5c850-901c-4e94-b395-5f70b9ad73bc.png">

It looks like `react-grid-layout` helpfully has some utils with the right math in them. Sadly they aren't exported for direct use, but hopefully you could use them to tweak the calculations you have here and in `handleWidthChange`

https://github.com/react-grid-layout/react-grid-layout/blob/master/lib/calculateUtils.js#L14-L35

Reply:
oh thanks i forgot to include the margin for each column when calculating the item width!